src/workflow/ArcanistDiffWorkflow.php

```diff
@@ -574,8 +574,27 @@ public function run() {
       if ($this->getArgument('plan-changes')) {
           $revision['fields']['plan-changes'] = true;
       }
+      $wip = (bool)preg_match('/\bwip\b/i', $revision['fields']['title']);
       // UBER CODE END
       if ($commit_message->getRevisionID()) {
+        // UBER CODE
+        if ($wip && !idx($revision['fields'], 'plan-changes', false)) {
+          // fetch revision status
+          $rev = $this->getConduit()->callMethodSynchronous(
+          'differential.query',
+          array(
+            'ids' => array($commit_message->getRevisionID()),
+          ));
+          $rev = head($rev);
+          if (idx($rev, 'statusName', false) == 'Changes Planned') {
+            $this->writeInfo('Changes Planned', phutil_console_format(
+              pht('Keyword "wip" detected and Differential Revision D%s is in '.
```

Comment:
I'd like to empathize "wip" was found on "title". Maybe better something like: 
`Keyword "wip" found in revision title and D%s is in...`

Reply:
New message would be:
```
Keyword "wip" found in title and Differential Revision D%s is in <fg:green>Changes Planned</fg> status, keeping same status
```